lib/http2/connection.c

```diff
@@ -66,6 +74,19 @@ static int foreach_request(h2o_context_t *ctx, int (*cb)(h2o_req_t *req, void *c
 
 const h2o_protocol_callbacks_t H2O_HTTP2_CALLBACKS = {initiate_graceful_shutdown, foreach_request};
 
+static int minimize_send_window(h2o_socket_t *sock)
+{
+#if !defined(TCP_NOTSENT_LOWAT)
+    return -1;
+#else
+    int fd = h2o_socket_get_fd(sock);
+    if (fd == -1)
+        return -1;
+    int val = 1; /* linux ignores the value if set to zero */
+    return setsockopt(fd, IPPROTO_TCP, TCP_NOTSENT_LOWAT, &val, sizeof(val));
```

Comment:
My reading of the `TCP_NOTSENT_LOWAT` patch (https://lwn.net/Articles/560082/) is that setting 1 as a value means that `epoll()` will only call `EPOLLOUT` if there are `0` unsent bytes on the socket. Is that the intended behavior? In the email, Eric Dumazet seemed to hint that using 128K would allow most applications to save memory (traded for CPU) without affecting throughput.